### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Category.Ulift

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
This meets all the CI requirements but it is a bit of dark magic with universes and Lean's universe unification. Lean 4 core instituted a hard error for unused universe parameters which blocks the existing construction `HomLift`.  

Short of a substantial change to the original file, I don't see any way to sidestep this error other than what is done here: make the unused universe parameter used in a trivial way. 

This should be ok for the process of porting but it should be revisited once porting is not a priority.

File: Mathlib/CategoryTheory/Category/ULift.lean [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+universe w₁ v₁ v₂ u₁ u₂
+
+namespace CategoryTheory
+
+variable {C : Type u₁} [Category.{v₁} C]
+
+/-- The functorial version of `ULift.up`. -/
+@[simps]
+def Ulift.upFunctor : C ⥤ ULift.{u₂} C where
```
Comment (adamtopaz) [inline Mathlib/CategoryTheory/Category/ULift.lean:54]: I suppose all the `Ulift`'s in this file should be `ULift` to match the naming in `Data.ULift`.

Comment (mattrobball) [reply Mathlib/CategoryTheory/Category/ULift.lean]: Done.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**